sdk/keyvault/keyvault-keys/src/lro/delete/operation.ts

```diff
@@ -2,110 +2,127 @@
    */
-  requestOptions?: RequestOptionsBase;
+  private async deleteKey(name: string, options: DeleteKeyOptions = {}): Promise<DeletedKey> {
+    const requestOptions = operationOptionsToRequestOptionsBase(options);
+    const span = createSpan("deleteKey", requestOptions);
+
+    let response: DeleteKeyResponse;
+    try {
+      response = await this.client.deleteKey(
+        this.vaultUrl,
+        name,
+        setParentSpan(span, requestOptions)
+      );
+    } finally {
+      span.end();
+    }
+
+    return getKeyFromKeyBundle(response);
+  }
+
   /**
-   * An interface representing a KeyClient. For internal use.
+   * The getDeletedKey method returns the specified deleted key along with its properties.
+   * This operation requires the keys/get permission.
+   * @summary Gets the specified deleted key.
+   * @param {string} name The name of the key.
+   * @param {GetDeletedKeyOptions} [options] The optional parameters.
    */
-  client: KeyClientInterface;
-}
+  private async getDeletedKey(
+    name: string,
+    options: GetDeletedKeyOptions = {}
+  ): Promise<DeletedKey> {
+    const responseOptions = operationOptionsToRequestOptionsBase(options);
+    const span = createSpan("getDeletedKey", responseOptions);
 
-/**
- * An interface representing a delete key's poll operation
- */
-export interface DeleteKeyPollOperation
-  extends PollOperation<DeleteKeyPollOperationState, DeletedKey> {}
-
-/**
- * @summary Reaches to the service and updates the delete key's poll operation.
- * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
- */
-async function update(
-  this: DeleteKeyPollOperation,
-  options: {
-    abortSignal?: AbortSignalLike;
-    fireProgress?: (state: DeleteKeyPollOperationState) => void;
-  } = {}
-): Promise<DeleteKeyPollOperation> {
-  const state = this.state;
-  const { name, client } = state;
+    let response: GetDeletedKeyResponse;
+    try {
+      response = await this.client.getDeletedKey(
+        this.vaultUrl,
+        name,
+        setParentSpan(span, responseOptions)
+      );
+    } finally {
+      span.end();
+    }
 
-  const requestOptions = state.requestOptions || {};
-  if (options.abortSignal) {
-    requestOptions.abortSignal = options.abortSignal;
+    return getKeyFromKeyBundle(response);
   }
 
-  if (!state.isStarted) {
-    const deletedKey = await client.deleteKey(name, requestOptions);
-    state.isStarted = true;
-    state.result = deletedKey;
-    if (!deletedKey.properties.recoveryId) {
-      state.isCompleted = true;
+  /**
+   * @summary Reaches to the service and updates the delete key's poll operation.
+   * @param [options] The optional parameters, which are an abortSignal from @azure/abort-controller and a function that triggers the poller's onProgress function.
+   */
+  public async update(
```

Comment:
This pattern of managing the client seems pretty nice, and definitely less leaky than before. It feels like a good balance between ruthless abstraction and the levels of duplication that we had previously, but it doesn't seem like it's much less verbose than what we had before.

It still feels bizarre to me that we have this state machine model where `update` can return a different operation, but I can't think of anywhere that we use it. That API suggests to me that returning a different kind of operation should be the way that we make the pollers progress in state, but instead we just have monolithic `update` functions that handle everything and then `return this`.

Reply:
I understand how this might be weird! I wonder if it would make sense to open an issue to have a discussion over time of how we could improve this in core-lro 🤔

Reply:
Yeah I think we should consider if we want this to be stateless or not (no strong feelings from me) and either we lean fully into stateless (get rid of `this`) or we lean fully away from it (don't bother returning `this`, because caller already has a copy of us)